server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -1748,9 +1752,10 @@ public boolean canCache(ShardSearchRequest request, SearchContext context) {
         if (request.requestCache() == null) {
             if (settings.getValue(IndicesRequestCache.INDEX_CACHE_REQUEST_ENABLED_SETTING) == false) {
                 return false;
-            } else if (context.size() != 0) {
+            } else if (context.size() != 0 && !canCacheSizeNonzeroRequests) {
```

Comment:
1. Can't this just be combined with the previous boolean conditional?
2. Should we use > 0? (Probably doesn't matter.)

Reply:
Pushed a commit with these changes.